rs-matter/src/dm/subscriptions.rs

```diff
@@ -70,32 +70,56 @@ impl Subscription {
     }
 }
 
+struct SubscriptionData<const N: usize> {
+    next_subscription_id: u32,
+    subscriptions: crate::utils::storage::Vec<Subscription, N>,
+}
+
+impl<const N: usize> SubscriptionData<N> {
+    /// Create the instance.
+    #[inline(always)]
+    const fn new() -> Self {
+        Self {
+            next_subscription_id: 1,
+            subscriptions: crate::utils::storage::Vec::new(),
+        }
+    }
+
+    /// Create an in-place initializer for the instance.
+    fn init() -> impl Init<Self> {
+        init!(Self {
+            next_subscription_id: 1,
+            subscriptions <- crate::utils::storage::Vec::init(),
+        })
+    }
+}
+
 /// A utility for tracking subscriptions accepted by the data model.
 ///
 /// The `N` type parameter specifies the maximum number of subscriptions that can be tracked at the same time.
 /// Additional subscriptions are rejected by the data model with a "resource exhausted" IM status message.
-pub struct Subscriptions<const N: usize = DEFAULT_MAX_SUBSCRIPTIONS> {
-    next_subscription_id: AtomicU32,
-    subscriptions: RefCell<crate::utils::storage::Vec<Subscription, N>>,
-    pub(crate) notification: Notification<NoopRawMutex>,
+pub struct Subscriptions<const N: usize = DEFAULT_MAX_SUBSCRIPTIONS, M = CriticalSectionRawMutex>
```

Comment:
My only a tad stronger than a nit request is to not default to `CriticalSectionRawMutex` but to `NoopRawMutex` instead, which saves a few pico-seconds if `rs-matter` would be used in a single-threaded context - which should be the majority of the cases and due to other places where we hard-code naked cells or `NoopRawMutex` is anyway the only possibility ATM until #258 is solved.

Reply:
Sweet, sounds good, thanks! Done